apps/checkout/src/hooks/useErrorMessages.ts

```diff
@@ -1,10 +1,29 @@
+import { ValidationErrorCode } from "@/lib/globalTypes";
 import { useFormattedMessages } from "./useFormattedMessages";
 
 export const useErrorMessages = () => {
   const formatMessage = useFormattedMessages();
 
-  return {
+  const errorMessages = {
     invalidValue: formatMessage("invalid"),
-    requiredField: formatMessage("required"),
+    requiredValue: formatMessage("required"),
+  };
+
+  const getMessageByErrorCode = (errorCode: ValidationErrorCode) => {
+    switch (errorCode) {
+      case "required":
+        return errorMessages.requiredValue;
+
+      case "invalid":
+        return errorMessages.invalidValue;
+
+      default:
+        break;
+    }
+  };
+
+  return {
+    errorMessages,
+    getMessageByErrorCode,
```

Comment:
Why not do something like the following?:
```
if code in ValidationErrorCode:
  return formatMessage(code);
else:
  return;
```

Reply:
^instead of manually mapping each code to each message

Reply:
We would have to do it like ```return formatMessage(`${code}Value`);``` where we could potentially add messages that don't have `Value` in them, then on the other hand I'd rather not shorten the messages to `invalid` or `required` since there may be e.g `required field` and `required` messages separately so.. yeah

Reply:
Also since there are not so many I don't think that this mapping is an overkill, In case there are much more I'll add some mapping util